plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-notices.php

```diff
@@ -84,27 +122,93 @@ public function add_paypal_migration_notice() {
 	}
 
 	/**
-	 * Add notice warning about the migration to PayPal Payments on the Payments settings page.
+	 * Add notice warning about PayPal account restriction.
 	 *
 	 * @return void
 	 */
-	public function add_paypal_migration_notice_on_payments_settings_page() {
-		global $current_tab, $current_section;
-		$is_payments_settings_page = 'woocommerce_page_wc-settings' === get_current_screen()->id && 'checkout' === $current_tab && empty( $current_section );
+	private function add_paypal_account_restricted_notice() {
+		// Skip if there's no account restriction flag.
+		if ( ! $this->has_account_restriction_flag() ) {
+			return;
+		}
 
-		// Only add the notice from this callback on the payments settings page.
-		if ( ! $is_payments_settings_page ) {
+		// Skip if the notice has been dismissed.
+		if ( $this->is_notice_dismissed( self::PAYPAL_ACCOUNT_RESTRICTED_NOTICE ) ) {
 			return;
 		}
-		$this->add_paypal_migration_notice();
+
+		$support_url = 'https://www.paypal.com/smarthelp/contact-us';
+		$dismiss_url = $this->get_dismiss_url( self::PAYPAL_ACCOUNT_RESTRICTED_NOTICE );
+		$message     = sprintf(
+			/* translators: 1: opening <a> tag, 2: closing </a> tag */
+			esc_html__( 'Your PayPal account has been restricted by PayPal. This may prevent customers from completing payments. Please %1$scontact PayPal support%2$s to resolve this issue and restore full functionality to your account.', 'woocommerce' ),
+			'<a href="' . esc_url( $support_url ) . '" target="_blank" rel="noopener noreferrer">',
+			'</a>',
+		);
+
+		$notice_html = '<div class="notice notice-error is-dismissible">'
+			. '<a class="woocommerce-message-close notice-dismiss" style="text-decoration: none;" href="' . esc_url( $dismiss_url ) . '"></a>'
+			. '<p><strong>' . esc_html__( 'PayPal Account Restricted', 'woocommerce' ) . '</strong></p>'
+			. '<p>' . $message . '</p>'
+			. '</div>';
+
+		echo wp_kses_post( $notice_html );
+	}
+
+	/**
+	 * Get the dismiss URL for a notice.
+	 *
+	 * @param string $notice_name The name of the notice.
+	 * @return string
+	 */
+	private function get_dismiss_url( $notice_name ) {
+		return wp_nonce_url(
+			add_query_arg( 'wc-hide-notice', $notice_name ),
+			'woocommerce_hide_notices_nonce',
+			'_wc_notice_nonce'
+		);
 	}
 
 	/**
-	 * Check if the installation notice has been dismissed.
+	 * Check if the notice has been dismissed.
 	 *
+	 * @param string $notice_name The name of the notice.
 	 * @return bool
 	 */
-	protected static function paypal_migration_notice_dismissed() {
-		return get_user_meta( get_current_user_id(), 'dismissed_paypal_migration_completed_notice', true );
+	private function is_notice_dismissed( $notice_name ) {
```

Comment:
Nit: another situation where explicit types would be nice. Especially because `get_user_meta()` will not return a boolean in many situations. 😬 
```suggestion
	private function is_notice_dismissed( string $notice_name ): bool {
```